src/views/Home.vue

```diff
@@ -94,22 +94,33 @@ export default {
       let chains = [...new Set(message.confirmations.map(c => c.chain))];
       return `${message.confirmations.length} confirmations:\n${chains.join(', ')}`;
     },
-    pushToMessageQueue ({data}) {
-      if(data){
-        try {
-          const message = JSON.parse(data)
-          this.last_messages.unshift(message)
-          if (this.last_messages.length > QUEUE_SIZE)
-            this.last_messages.pop()
-        } catch (err) {
-          console.log(err)
-        }
-      }
+    pushToMessageQueue (data) {
+      const message = data
+      this.last_messages.unshift(message)
+      this.last_messages.pop()
```

Comment:
but if the last_messages started out with less than QUEUE_SIZE items, wouldn't we unnecessarily pop items?

Reply:
Alright, overlooked the eventListener defintion in line 107.